src/sagemaker/jumpstart/converter_utils.py

```diff
@@ -0,0 +1,105 @@
+# Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"). You
+# may not use this file except in compliance with the License. A copy of
+# the License is located at
+#
+#     http://aws.amazon.com/apache2.0/
+#
+# or in the "license" file accompanying this file. This file is
+# distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF
+# ANY KIND, either express or implied. See the License for the specific
+# language governing permissions and limitations under the License.
+"""This module stores CuratedHub converter utilities for JumpStart."""
+from __future__ import absolute_import
+
+import re
+from typing import Any, Dict, List
+from sagemaker.jumpstart.enums import ModelSpecKwargType, NamingConventionType
+
+
+def camel_to_snake(camel_case_string: str) -> str:
+    """Converts camelCaseString or UpperCamelCaseString to snake_case_string."""
+    snake_case_string = re.sub("(.)([A-Z][a-z]+)", r"\1_\2", camel_case_string)
+    return re.sub("([a-z0-9])([A-Z])", r"\1_\2", snake_case_string).lower()
+
+
+def snake_to_upper_camel(snake_case_string: str) -> str:
+    """Converts snake_case_string to UpperCamelCaseString."""
+    upper_camel_case_string = "".join(word.title() for word in snake_case_string.split("_"))
+    return upper_camel_case_string
+
+
+def walk_and_apply_json(json_obj: Dict[Any, Any], apply):
+    """Recursively walks a json object and applies a given function to the keys."""
+
+    def _walk_and_apply_json(json_obj, new):
+        if isinstance(json_obj, dict):
+            if isinstance(new, dict):
+                for key, value in json_obj.items():
+                    new_key = apply(key)
+                    if isinstance(value, dict):
+                        new[new_key] = {}
+                        _walk_and_apply_json(value, new=new[new_key])
```

Comment:
too much indentation here, can you refactor  so it's easier to read?

Reply:
done